momepy/functional/_diversity.py

```diff
@@ -176,6 +188,322 @@ def describe(
     return _compute_stats(grouper, q, include_mode)
 
 
+def values_range(
+    data: DataFrame | Series, graph: Graph, rng: tuple | list = (0, 100), **kwargs
+):
+    """Calculates the range of values within neighbours defined in ``graph``.
+    Uses ``scipy.stats.iqr`` under the hood.
+
+    Adapted from :cite:`dibble2017`.
+
+    Parameters
+    ----------
+    data : DataFrame | Series
+        A DataFrame or Series containing the values to be analysed.
+    graph : libpysal.graph.Graph
+        A spatial weights matrix for the data.
+    rng : tuple, list, optional (default (0,100)))
+        A two-element sequence containing floats between 0 and 100 (inclusive)
+        that are the percentiles over which to compute the range.
+        The order of the elements is not important.
+    **kwargs : dict
+        Optional arguments for ``scipy.stats.iqr``.
+
+    Returns
+    ----------
+    Series
+        A Series containing resulting values.
+
+    Examples
+    --------
+    >>> tessellation_df['area_IQR_3steps'] = mm.range(tessellation_df['area'],
+    ...                                               graph,
+    ...                                               rng=(25, 75))
+    """
+
+    def _apply_range(values):
+        return sp.stats.iqr(values, rng=rng, **kwargs)
```

Comment:
Might be actually easier to use numpy.percentile directly like we have in describe and pass the kwargs there for the sake of consistency. you can also njit that.